internal/api/connectors/inventory/inventory.go

```diff
@@ -0,0 +1,190 @@
+package inventory
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"playbook-dispatcher/internal/common/constants"
+	"playbook-dispatcher/internal/common/utils"
+	"time"
+
+	"github.com/redhatinsights/platform-go-middlewares/request_id"
+	"github.com/spf13/viper"
+)
+
+const basePath = "/api/inventory/"
+
+type InventoryConnector interface {
+	GetHostDetails(
+		ctx context.Context,
+		IDs []string,
+		orderBy string,
+		orderHow string,
+		limit int,
+		offest int,
+	) (hostDetails []HostDetails, err error)
+
+	GetSystemProfileDetails(
+		ctx context.Context,
+		IDs []string,
+		orderBy string,
+		orderHow string,
+		limit int,
+		offset int,
+	) (profileDetails []SystemProfileDetails, err error)
+}
+
+type inventoryConnectorImpl struct {
+	client ClientWithResponsesInterface
+}
+
+func createHostGetHostByIdParams(orderBy string, orderHow string) *ApiHostGetHostByIdParams {
+	orderByParam := OrderByParam(orderBy)
+	orderHowParam := OrderHowParam(OrderHowParam_ASC)
+
+	return &ApiHostGetHostByIdParams{
+		OrderBy:  &orderByParam,
+		OrderHow: &orderHowParam,
+	}
+}
+
+func createHostGetHostSystemProfileByIdParams(orderBy string, orderHow string) *ApiHostGetHostSystemProfileByIdParams {
+	orderByParam := OrderByParam(orderBy)
+	orderHowParam := OrderHowParam(orderHow)
+	fields := FieldsParam(
+		SystemProfileNestedObject{
+			AdditionalProperties: map[string]interface{}{
+				"fields[system_profile]": []string{"rhc_client_id", "owner_id"},
+			},
+		},
+	)
+
+	return &ApiHostGetHostSystemProfileByIdParams{
+		OrderBy:  &orderByParam,
+		OrderHow: &orderHowParam,
+		Fields:   &fields,
+	}
+}
+
+func createHostDetails(response *ApiHostGetHostByIdResponse) []HostDetails {
+	hostDetails := make([]HostDetails, len(response.JSON200.Results))
+
+	for i, response := range response.JSON200.Results {
+		satelliteFacts := getSatelliteFacts(response.Facts)
+
+		hostDetails[i] = HostDetails{
+			ID:          *response.Id,
+			DisplayName: *response.DisplayName,
+			Facts:       satelliteFacts,
+			CanonicalFacts: map[string]interface{}{
+				"fqdn": *response.CanonicalFactsOut.Fqdn,
+			},
+		}
+	}
+
+	return hostDetails
+}
+
+func createSystemProfileDetails(response *ApiHostGetHostSystemProfileByIdResponse) []SystemProfileDetails {
+	systemProfileDetails := make([]SystemProfileDetails, len(response.JSON200.Results))
+	for i, response := range response.JSON200.Results {
+		systemProfileDetails[i] = SystemProfileDetails{
+			ID: *response.Id,
+			SystemProfileFacts: map[string]interface{}{
+				"rhc_client_id": *response.SystemProfile.RhcClientId,
```

Comment:
It might be possible for the system profile pointer to be nil and/or the data in the system profile to be nil.

Here's a test case that shows how this could happen:

```
		It("Interperates response correctly when no system profile is provided", func() {
			doer := test.MockHttpClient(200, `{"results":[{"id":"1234"}]}`)

			client := NewInventoryClientWithHttpRequestDoer(config.Get(), &doer)
			ctx := test.TestContext()
			IDs := []string{"1234"}
			_, err := client.GetSystemProfileDetails(ctx, IDs, "DisplayName", "ASC", 10, 0)
			Expect(err).To(HaveOccurred())
			Expect(err.Error()).To(ContainSubstring(`unexpected status code "400"`))
		})

		It("Interperates response correctly when system proflie does not include rhc_client_id", func() {
			doer := test.MockHttpClient(200, `{"results":[{"id":"1234","system_profile":{"owner_id":"b2ea37a0-7fb0-4f14-815d-fb582a916d5b"}}]}`)

			client := NewInventoryClientWithHttpRequestDoer(config.Get(), &doer)
			ctx := test.TestContext()
			IDs := []string{"1234"}
			_, err := client.GetSystemProfileDetails(ctx, IDs, "DisplayName", "ASC", 10, 0)
			Expect(err).To(HaveOccurred())
			Expect(err.Error()).To(ContainSubstring(`unexpected status code "400"`))
		})
```